src/node/miner.h

```diff
@@ -203,9 +202,6 @@ int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParam
 
 /** Update an old GenerateCoinbaseCommitment from CreateNewBlock after the block txs have changed */
 void RegenerateCommitments(CBlock& block, ChainstateManager& chainman);
-
-/** Apply -blockmintxfee and -blockmaxweight options from ArgsManager to BlockAssembler options. */
-void ApplyArgsManOptions(const ArgsManager& gArgs, BlockAssembler::Options& options);
```

Comment:
In commit "rpc: call CreateNewBlock via miner interface" (f375fb9a0dff7ccead80626f4aee853db7799e6f)

It might be good to keep this ApplyArgsManOptions function for more consistency with other options parsing code. The node/interfaces.cpp createNewBlock function could call ApplyArgsManOptions instead of parsing options itself. The current approach also seems ok though if you prefer it.

Reply:
I split f375fb9a0dff7ccead80626f4aee853db7799e6f in two parts:

1. 6d6ddcb759465c7375ff772faa960cf92805468a simply adds and uses the interface
2. 6b7c8c3e27ae529306d18019e7bd94b4ae5b9a21 drops the `BlockAssembler` constructor without `options` argument

I changed this second commit to keep `ApplyArgsManOptions`.